### PR TITLE
fix: export sliced array child rows to Arrow

### PR DESCRIPTION
**Pull request by taiyang-li**
### What problem does this PR solve?

Fix Arrow export for sliced Array/Map vectors when the compact offsets fast path is used.

  ArrayVector/MapVector::slice keeps the original child vector and only slices parent offsets/sizes. In
  the normal Arrow export path, exportOffsets() previously copied compact parent offsets but left
  childRows as the default full-child selection. As a result, exporting a sliced Array/Map could still
  export the full child vector. For nested VARCHAR/VARBINARY, this could make the exported child string
  buffer exceed Arrow's int32 offset limit even when the top-level batch had already been sliced.

  This change rebases compact offsets to the current slice and restricts childRows to the referenced
  child range [base, end), matching the behavior already used by exportOffsetsIPC().

### Type of Change

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🚀 Performance improvement (optimization)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🔨 Refactoring (no logic changes)
- [ ] 🔧 Build/CI or Infrastructure changes
- [ ] 📝 Documentation only

### Description

Describe your changes in detail.
For complex logic, explain the "Why" and "How".

### Performance Impact

- [x] **No Impact**: This change does not affect the critical path (e.g., build system, doc, error handling).
- [ ] **Positive Impact**: I have run benchmarks.
- [ ] **Negative Impact**: Explained below (e.g., trade-off for correctness).

### Release Note

Please describe the changes in this PR

Release Note:

```text
Release Note:
- Fix Arrow export for sliced Array/Map vectors when the compact offsets fast path is used.
```

### Checklist (For Author)

- [x] I have added/updated unit tests (ctest).
- [x] I have verified the code with local build (Release/Debug).
- [x] I have run clang-format / linters.
- [ ] (Optional) I have run Sanitizers (ASAN/TSAN) locally for complex C++ changes.
- [ ] No need to test or manual test.

### Breaking Changes

- [x] No
- [ ] Yes (Description: ...)
    <details>
    <summary>Click to view Breaking Changes</summary>

    ```text
    Breaking Changes:
    - Description of the breaking change.
    - Possible solutions or workarounds.
    - Any other relevant information.
    ```
    </details>

### CODE REVIEW

File: bolt/vector/arrow/tests/ArrowBridgeArrayTest.cpp [hunk 1]
```diff
@@ -713,6 +713,28 @@ TEST_F(ArrowBridgeArrayExportTest, arraySimple) {
   testArrayVector(data3);
 }
 
+TEST_F(
+    ArrowBridgeArrayExportTest,
+    slicedArrayStringExportsOnlyReferencedElements) {
+  auto elements = vectorMaker_.flatVector<StringView>(
+      {"unused-prefix", "a", "b", "c", "d", "unused-suffix"});
+  auto offsets = makeBuffer<vector_size_t>({0, 1, 3, 4});
```
Comment (guhaiyan0221) [inline bolt/vector/arrow/tests/ArrowBridgeArrayTest.cpp:721]: Could we add a test for sliced Array/Map export with null parent rows? The compact fast path now exports a continuous child range [base, end), while the slow path skips null parent rows when building childRows. A test would help ensure null parent offsets/sizes don't cause unused child values to be exported.

Comment (taiyang-li) [reply bolt/vector/arrow/tests/ArrowBridgeArrayTest.cpp:721]: @guhaiyan0221 I had improved it and added another 2 uts, pls help review the latest commit, thanks!

**Review by guhaiyan0221 — APPROVED** [review]
